Add garrysmod addon

luttje opened the pull request:
Closes #48

Adds addon with definitions for the game [Garry's Mod](https://gmod.facepunch.com/). The definitions are automatically generated and based on [the official wiki](https://wiki.facepunch.com/gmod)

The submodule points to the `lua-language-server-addon` branch in [this repo](https://github.com/luttje/glua-api-snippets/tree/lua-language-server-addon)

carsakiller commented:
Please run [StyLua](https://github.com/JohnnyMorganz/StyLua) to apply the proper code style.

luttje commented:
Thanks for the swift check! I see now that the instruction to do that is only in the 'New PR'-template. I missed that since I created this PR through the 'compare changes' suggestion. **Perhaps I can also help by adding an instruction for that in the contributions guide?**

In any case I have implemented the running of StyLua in the workflow that generates this addon and I've updated the addon now. Please let me know if there's anything else I can do.

carsakiller commented:
Ah, good catch, thanks! I'll add that to the publishing steps. Good work!
